geoh5py/groups/property_group.py

```diff
@@ -270,33 +234,44 @@ def property_group_type(self) -> GroupTypeEnum:
         """
         return self._property_group_type
 
-    @property_group_type.setter
-    def property_group_type(self, value: str | GroupTypeEnum | None):
-        if (
-            getattr(self, "_property_group_type", GroupTypeEnum.UNKNOWN)
-            is not GroupTypeEnum.UNKNOWN
-        ):
-            raise UserWarning(
-                "Cannot modify 'property group type' of an existing property group. "
-                "Consider creating a new property group."
-            )
+    def reference_to_data(self, data: str | Data | UUID) -> Data:
```

Comment:
I have the feeling this could be a nice utils function no?
just passing parent as an attribute

Reply:
I think we could move it to the ObjectBase actually, and just call the `parent.reference_to_data`? I will let you modify it on your PR if you want.